docs/docs/components/service-mesh.md

```diff
@@ -14,6 +14,20 @@ all workloads which have a specified configuration. The service mesh container f
 sets up `iptables` rules based on its configuration and then starts
 [Envoy](https://www.envoyproxy.io/) for TLS origination and termination.
 
+## Service mesh startup enforcement
+Since Contrast doesn't yet enforce the order in which the containers are started
+(see [Limitations](../features-limitations.md)), we deny all incoming connections
+until the service mesh resolves this deny rule.
+A systemd unit inside the podVM creates this deny rule.
+The kata-agent systemd unit requires that this unit successfully runs and exists.
+Therefore, the deny rule is in place, before any containers can be started.
+
+Contrast ensures that this deny rule doesn't drop traffic in scenarios that don't
+require a service mesh by injecting the service mesh container not as a sidecar
+but as an init container. During generation, the CLI also sets `CONTRAST_SERVICE_MESH_DISABLE`
+as an environment on the container to disable any further service mesh setup.
+The container just removes the deny rule and exits.
```

Comment:
Did you consider lifting the deny rule in the initializer instead? That would avoid the need for an additional image pull.

Reply:
Yes, but then we'd also need NET_ADMIN for the initializer. Also the coordinator pod, who doesn't have an initializer (yet) needs either NET_ADMIN to remove the rule itself or needs an "disabled" initializer. In summary, using only/one additional initializer might be optimal. Though I have no string favorite. Do you have a preference?

Reply:
The Coordinator does have `SYS_ADMIN`, and I would not mind the `NET_ADMIN` too much in a real (terminating) init container. But I see the point of not duplicating the removal rule between the coordinator and the initializer. Do I understand correctly that the Coordinator now gets a (disabled) mesh sidecar, too? I missed that in review :sweat_smile:

Reply:
> understand correctly that the Coordinator now gets a (disabled) mesh sidecar, too?
yes. The Coordinator currently doesn't have NET_ADMIN and is a long-running (albeit a highly trusted) component. Therefore, I didn't want to add it, if it's not needed in the long term. But I think it's a discussion of a new container pull for _every_ confidential pod or _one_ additional image pull for the coordinator. So we should go with option 2, right? This is also the reason why the defaultdeny package is already in the shared /internal folder, because I first implemented it in the initializer.

Reply:
The SYS_ADMIN cap should include NET_ADMIN, I think? 

I don't understand the options you're mentioning - afaiu we have either

1. Add defaultdeny lifting to the initializer (conditional on service mesh absence) and the coordinator (unconditional).
2. Add service mesh container to the coordinator and all non-mesh pods.

The more I think about it, the more I prefer (1).